core/src/main/scala/org/apache/spark/ui/JettyUtils.scala

```diff
@@ -274,25 +277,28 @@ private[spark] object JettyUtils extends Logging {
       conf: SparkConf,
       serverName: String = ""): ServerInfo = {
 
-    val collection = new ContextHandlerCollection
     addFilters(handlers, conf)
 
     val gzipHandlers = handlers.map { h =>
+      h.setVirtualHosts(Array("@" + SPARK_CONNECTOR_NAME))
```

Comment:
Do we need this code here? `setVirtualHosts` should always be called in `addHandler` for each handler right?